### PR TITLE
Basic housekeeping of riak/session_store and ripple/session_store

### PR DESCRIPTION
**Pull request by brvtl**
Just some housekeeping

P.S. Hi, I don't know much about Riak but quite enjoyed a talk I saw on it at ScaleConf in South Africa so I thought I would make an effort to contribute. I've started reading through some of the ruby source and have made some very minor changes. If I'm overlooking something please set me straight - but for now I'm forced to use the tests as a guideline.

### CODE REVIEW

File: lib/ripple/session_store.rb [hunk 1]
```diff
@@ -51,20 +51,22 @@ def set_session(env, sid, session_data)
     end
 
     def destroy(env)
-      if sid = current_session_id(env)
-        @bucket.delete(sid)
-      end
+      @bucket.delete current_session_id(env)
```
Comment (seancribbs) [inline lib/ripple/session_store.rb:54]: If the session ID is nil, you don't want to delete it.

File: lib/riak/session_store.rb [hunk 2]
```diff
@@ -72,16 +69,12 @@ def set_session(env, session_id, session, options)
...
         # Rails 2.3 kills the session id from the request when
         # reset_session is called. Working around that by temp.
         # storing it in the middleware and explicitly destroying it
         # as it's not guaranteed that Riak expiry is enabled
-        session_id = if @session_id
-                       destroy_session(env, @session_id, options)
-                     else
-                       generate_sid
-                     end
+        session_id = @session_id ? destroy_session(env, @session_id, options) : generate_sid
```
Comment (seancribbs) [inline lib/riak/session_store.rb:77]: This is stylistic, meh.

File: lib/riak/session_store.rb [hunk 3]
```diff
@@ -54,16 +54,13 @@ def generate_sid
...
-      if session_id && robject = (bucket.get(session_id) rescue nil)
-        @session_id = session_id
-        if stale?(robject)
-          bucket.delete(session_id)
-          fresh_session
-        else
-          [session_id, robject.data]
-        end
-      else
+      return fresh_session unless session_id && robject = (bucket.get(session_id) rescue nil)
```
Comment (seancribbs) [inline lib/riak/session_store.rb:57]: I don't see how this is better.

File: lib/ripple/session_store.rb (hunk 1 above)
Comment (brvtl) [inline lib/ripple/session_store.rb:54]: Ah thank you - that was an oversight on my behalf. I'll add a spec.

File: lib/riak/session_store.rb (hunk 2 above)
Comment (brvtl) [inline lib/riak/session_store.rb:77]: Eh, maybe I have a fetish for them

**Pull request closed without merging**